### PR TITLE
Backend to microservice.

### PR DESCRIPTION
**Pull request by parc-jason**
# Description
Changes for moving backend into microservice.
Need to update the openshift templates and Jenkinsfile.
Updates to the responses, they were standardized and moved into microservice.

## Types of changes
Move backend into another repo specifically for providing a microservice.

Bug fix (non-breaking change which fixes an issue)
New feature (non-breaking change which adds functionality)
Documentation (non-breaking change with enhancements to documentation)
Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -41,6 +41,14 @@ pipeline {
     PROD_HOST = "${APP_NAME}.${APP_DOMAIN}"
     // will be added to the HOST_ROUTE
     PATH_ROOT = "/${APP_NAME}"
+
+    // for the email microservice backend
+    EMAIL_MICROSRV_BC = "https://raw.githubusercontent.com/bcgov/nr-email-microservice/1.0.0/openshift/api.bc.yaml"
+    EMAIL_MICROSRV_DC = "https://raw.githubusercontent.com/bcgov/nr-email-microservice/1.0.0/openshift/api.dc.yaml"
+    EMAIL_MICROSRV_REF = "1.0.0"
```
Comment (jujaga) [inline Jenkinsfile]: May be worth parameterizing the branch/tag name in the raw urls.

File: Jenkinsfile.cicd [hunk 2]
```diff
@@ -435,11 +408,14 @@ def deployStage(String stageEnv, String projectEnv, String hostEnv, String pathE
...
-            "REPO_NAME=${REPO_NAME}",
-            "JOB_NAME=${JOB_NAME}",
+            "${EMAIL_MICROSRV_DC}",
+            "APP_LABEL=${EMAIL_MICROSRV_APP_LABEL}",
+            "IMAGE_NAME=${EMAIL_MICROSRV_IMAGE_NAME}",
             "NAMESPACE=${projectEnv}",
-            "APP_NAME=${APP_NAME}"
+            "SECRET_NAME=cmsg-client",
+            "CONFIG_MAP_NAME=cmsg-config",
+            "CMSG_SENDER=NR.CommonServiceShowcase@gov.bc.ca",
```
Comment (jujaga) [inline Jenkinsfile.cicd]: Don't need to fix anything, but as a note, using single quote strings for things that do not need interpolation is the "groovy" way of handling strings, whereas things with templating and need interpolation need to use double quotes.

File: README.md [hunk 3]
```diff
@@ -1,12 +1,10 @@
 # Natural Resources Messaging Service Showcase
 
-This repository contains code to showcase the Common Messaging Service (CMSG).  It demonstrates how to authenticate use the CMSG API.  We will also provide a simple UI to integrate the CMSG capabilities into an application. The technologies we use for this example are: [Node.js](https://nodejs.org/) for the backend server (calls to CMSG) and [React.js](https://reactjs.org) for the frontend UI.
+This repository contains code to provide a simple UI for the [NR Email Microservice](https://github.com/bcgov/nr-email-microservice).  The Email Microservice provides an example implementation for use of the Common Messaging Service API (CMSG).  We [React.js](https://reactjs.org) for the frontend UI.
```
Comment (jujaga) [inline README.md]: We use React.js (missing use)

File: openshift/README.md [hunk 4]
```diff
@@ -1,8 +1,8 @@
...
+The Messaging Service Showcase application is quite simple.  It is a [node.js](https://nodejs.org/) API (see [NR Email Microservice](https://github.com/bcgov/nr-email-microservice)) and a [React.js](https://reactjs.org) UI (see [frontend](../frontend/README.md)).  In production, both the backend and frontend are placed behind a reverse proxy (see [reverse-proxy](../reverse-proxy/README.md)).
 
 
-For both the frontend and backend, we provide a set of npm scripts in their respective package.json files.  The most important of which are: backend "npm run start", and frontend is "npm run build".  The reverse proxy has no installation scripts (but does have runtime configuration requirements, see [../reverse-proxy](../reverse-proxy/README.md)).
+For both the frontend, we provide a set of npm scripts in [package.json](../frontend/package.json) file; the most important of which is: "npm run build".  The reverse proxy has no installation scripts (but does have runtime configuration requirements, see [../reverse-proxy](../reverse-proxy/README.md)).
```
Comment (jujaga) [inline openshift/README.md]: Backtick quotes for `npm run build` will make it stand out more.

**Comment by parc-jason** [comment]
Will address these changes in next PR

**Pull request merged**